util/fipstools/acvp/acvptool/subprocess/pbkdf.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2020, Google Inc.
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+// SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+// OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+// CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+package subprocess
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// The following structures reflect the JSON of ACVP PBKDF tests. See
+// https://pages.nist.gov/ACVP/draft-celi-acvp-pbkdf.html
+
+type pbkdf struct{}
+
+type pbkdfVectorSet struct {
+	Groups []pbkdfTestGroup `json:"testGroups"`
+}
+
+type pbkdfTestGroup struct {
+	ID      uint64 `json:"tgId"`
+	HmacAlg string `json:"hmacAlg"`
+	Type    string `json:"testType"`
+	Tests   []struct {
+		ID         uint64 `json:"tcId"`
+		KeyLen     uint64 `json:"keyLen"`
+		SaltHex    string `json:"salt"`
+		Pwd        string `json:"password"`
+		Iterations uint64 `json:"iterationCount"`
+	} `json:"tests"`
+}
+
+type pbkdfTestGroupResponse struct {
+	ID    uint64              `json:"tgId"`
+	Tests []pbkdfTestResponse `json:"tests"`
+}
+
+type pbkdfTestResponse struct {
+	ID            uint64 `json:"tcId"`
+	DerivedKeyHex string `json:"derivedKey"`
+}
+
+// List of supported HMAC Algorithms by NIST
+var HmacAlgList map[string]bool = map[string]bool{
+	"SHA-1":        true,
+	"SHA2-224":     true,
+	"SHA2-256":     true,
+	"SHA2-384":     true,
+	"SHA2-512":     true,
+	"SHA2-512/256": true,
```

Comment:
SHA2-512/256 isn't Approved in FIPS mode (see `PBKDF2_verify_service_indicator()` in `crypto/fipsmodule/service_indicator/service_indicator.c`).